### ISSUE TITLE
Compilation fails for dependencies with scope=provided

### ISSUE THREAD

**Issue by vegegoku** [Vertispan/j2clmavenplugin#36]
Steps to reproduce in dominodo sample app located at
https://github.com/DominoKit/dominodo/tree/dominodo-j2cl/dominodo

1- in dominodo-client change the dominodo-shared dependency scope provided.
2- in dominodo-j2cl add the dominodo-shared dependency with scope compile
3- in the root project execute `mvn clean install`

Compilation will fail, for some reason the dominodo-shared classes were not transpiled into js which make closure fails.

```
...
done with pool work for com.google.jsinterop:jsinterop-annotations:2.0.0 TranspileSources in 0ms
/home/vegegoku/.m2/gwt3BuildCache/0.13-SNAPSHOT/dominodo-j2cl-e4a55b100a8273d1869f6eff13fad2dc5111b526/closure-output-54378f2da51d7c5f4b0ba9db497ce0f4082d3e29/org/dominokit/domino/ui/forms/Select_Priority_SimpleBeanEditorDelegate.impl.java.js:7: ERROR - [JSC_MISSING_MODULE_OR_PROVIDE_FOR_FORWARD_DECLARE] Required namespace "org.dominokit.samples.Priority$impl" never defined.
let Priority = goog.forwardDeclare('org.dominokit.samples.Priority$impl');
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/vegegoku/.m2/gwt3BuildCache/0.13-SNAPSHOT/dominodo-j2cl-e4a55b100a8273d1869f6eff13fad2dc5111b526/closure-output-54378f2da51d7c5f4b0ba9db497ce0f4082d3e29/org/dominokit/domino/ui/forms/Select_Priority_SimpleBeanEditorDelegate.impl.java.js:65: ERROR - [JSC_MISSING_MODULE_OR_PROVIDE] Required namespace "org.dominokit.samples.Priority$impl" never defined.
  Priority = goog.module.get('org.dominokit.samples.Priority$impl');
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/vegegoku/.m2/gwt3BuildCache/0.13-SNAPSHOT/dominodo-j2cl-e4a55b100a8273d1869f6eff13fad2dc5111b526/closure-output-54378f2da51d7c5f4b0ba9db497ce0f4082d3e29/org/dominokit/domino/ui/forms/Select_Priority_SimpleBeanEditorDelegate.java.js:6: ERROR - [JSC_MISSING_MODULE_OR_PROVIDE] Required namespace "org.dominokit.samples.Priority" never defined.
goog.require('org.dominokit.samples.Priority');
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/vegegoku/.m2/gwt3BuildCache/0.13-SNAPSHOT/dominodo-j2cl-e4a55b100a8273d1869f6eff13fad2dc5111b526/closure-output-54378f2da51d7c5f4b0ba9db497ce0f4082d3e29/org/dominokit/domino/ui/forms/Select_Project_SimpleBeanEditorDelegate.impl.java.js:7: ERROR - [JSC_MISSING_MODULE_OR_PROVIDE_FOR_FORWARD_DECLARE] Required namespace "org.dominokit.samples.Project$impl" never defined.
let Project = goog.forwardDeclare('org.dominokit.samples.Project$impl');
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
```

**Comment by niloc132** [comment]
Consider #22 when addressing this.

**Comment by niloc132** [comment]
This should be fixed in 0.19-snapshot, but I haven't specifically confirmed it.

**Comment by vegegoku** [comment]
the reproducer here have been updated to latest j2cl-m-p and it seems to work now, it compiles and runs successfully in watch mode

https://github.com/DominoKit/dominodo/tree/dominodo-j2cl/dominodo
